### PR TITLE
[android_intent] Android Code Inspection and Clean up

### PR DESCRIPTION
**Pull request by hamdikahloun**
## Description

- Handle deprecation & unchecked warning as error

- Avoiding uses or overrides a deprecated API in FlutterBackgroundExecutor.java

## Related Issues

flutter/flutter#65970

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/MethodCallHandlerImpl.java [hunk 1]
```diff
@@ -75,19 +76,24 @@ public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
...
-        (!TextUtils.isEmpty(packageName)
-                && !TextUtils.isEmpty((String) call.argument("componentName")))
-            ? new ComponentName(packageName, (String) call.argument("componentName"))
-            : null;
+            (packageName != null
+                    && component != null
+                    && !TextUtils.isEmpty(packageName)
+                    && !TextUtils.isEmpty(component))
+                    ? new ComponentName(packageName, component)
+                    : null;
```
Comment (dnfield) [inline packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/MethodCallHandlerImpl.java:91]: nit: I would find this easier to read as a regular declaration followed by an if statement. This seems like a lot of logic for a ternary.

File: packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/MethodCallHandlerImpl.java [hunk 2]
```diff
@@ -146,42 +152,67 @@ private static Bundle convertArguments(Map<String, ?> arguments) {
         bundle.putLongArray(key, (long[]) value);
       } else if (value instanceof double[]) {
         bundle.putDoubleArray(key, (double[]) value);
-      } else if (isTypedArrayList(value, Integer.class)) {
-        bundle.putIntegerArrayList(key, (ArrayList<Integer>) value);
-      } else if (isTypedArrayList(value, String.class)) {
-        bundle.putStringArrayList(key, (ArrayList<String>) value);
-      } else if (isStringKeyedMap(value)) {
-        bundle.putBundle(key, convertArguments((Map<String, ?>) value));
+      } else if (integers(value) != null) {
```
Comment (dnfield) [inline packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/MethodCallHandlerImpl.java:155]: nit: I think I prefer the old name of these - I'm a bit surprised to see a function named `integers` or `strings` or `map`.

I also wonder if we can just somehow avoid traversing the array twice and creating the object twice.

**Review by dnfield — APPROVED** [review]
LGTM, but needs to have merge conflict(s) fixed up.

/cc @xster FYI as well here

**Comment by xster** [comment]
If it passes pre-submit then you're good

**Comment by hamdikahloun** [comment]
@dnfield 
merge conflict(s) fixed .
Thank you

**Pull request merged**